main.py

```diff
@@ -503,7 +510,15 @@ def main(cl_arguments):
     if args.do_target_task_training:
         # Train on target tasks
         pre_target_train_path = setup_target_task_training(args, target_tasks, model, strict)
-
+        # Check for previous target train checkpoints
+        task_directory, _, _ = check_for_previous_checkpoints(
+            args.run_dir, target_tasks, "target_train", args.load_model
+        )
+        if task_directory is not None:
+            # If there is a task to restore from, target train only on target tasks
+            # including and following that task.
+            last_task_index = [task.name for task in target_tasks].index(task_directory)
+            target_tasks = target_tasks[last_task_index:]
```

Comment:
Make sure this filtering operation doesn't impact the eval stage. We still want to eval on everything.